### ISSUE TITLE
Improve user documentation (feedback from students)

### ISSUE THREAD

**Issue by Carltoffel** [fortran-lang/stdlib#697]
I'm currently giving a Fortran lecture with twelve students, and today I introduced modules, fpm dependencies and finally the stdlib.

I didn't tell them much about the stdlib, instead I asked them to try it out by themselves. Most of them directly found the documentation, but they seemed to be a bit underwhelmed by it.

I'm not up-to-date with the progress of stdlib... sorry if I mention duplicate/obvious things.
Problems that occurred:
- missing or incomplete documentation
- not clear enough what *public* functions a module has (one student tried to use xoshiro256ss). I think it would be best to have a toggle for private functions
- navigating the docu: when searching in the documentation, it's easier to find the source file, which might be `.fypp` instead of the actual documentation which is a bit scary for students

After the initial frustration and confusion, they found many useful modules, e.g.:
- error_stop and check
- string type
- ascii
- bitsets
- io (npy)
- arange
- optval
- sorting
- stats

I think overall the routines convinced them, but the documentation definitely needs some improvement. That shouldn't be surprising.

**Comment by milancurcic** [comment]
Indeed. I don't think we have an actual user doc with explanations and examples, only the spec docs. Definitely much needed.

**Comment by jvdp1** [comment]
Thank you @Carltoffel for the feedback! Really nice that you tried stdlib!

Re: the documentation, @awvwgk set up a [project](https://github.com/awvwgk/stdlib-docs) a while ago for generating something like `fpm`, with tutorials, how-tos's, ....
Here is the generated [site](https://awvwgk.github.io/stdlib-docs/en/index.html), with an example for [stats](https://awvwgk.github.io/stdlib-docs/en/how-to/desc_stats.html) and I/O with [npy files](https://awvwgk.github.io/stdlib-docs/en/how-to/npy.html).

Would such a website have helped your students? Or would you expect something different?

**Comment by milancurcic** [comment]
> Re: the documentation, @awvwgk set up a project a while ago for generating something like fpm, with tutorials, how-tos's, ....
Here is the generated site, with an example for stats and I/O with npy files.

This is excellent, I either haven't seen it or have completely forgotten about it. I think that would be a great solution to build on.

**Comment by jvdp1** [comment]
Yes, maybe we should put first our effort in this documentation to attrack users for stdlib.

**Comment by awvwgk** [comment]
I started this after we made the fpm docs (https://github.com/fortran-lang/stdlib/discussions/599), but as so often, other things came up and I never got far enough that we could transfer it to fortran-lang.

**Comment by jalvesz** [comment]
Hi, I think this idea is excellent and a must! Good documentation is paramount for making the library really attractive. @awvwgk I just saw your draft for the site, it looks brilliant, is it based on Sphinx?

**Comment by awvwgk** [comment]
The website is based on sphinx, happy to give it a push to revive it again if there is interest from people to help with finishing the draft.

**Comment by jalvesz** [comment]
Yes, looking forward to give a hand with this project! Let me know how could I help.

**Comment by jvdp1** [comment]
@awvwgk what should be done to revive this? I guess that we should first update the section related to `fpm`, and probably add new contents in the section `how to`.

**Comment by jalvesz** [comment]
An "easy" win I think would be to move the specs under this project. At least to try out and see how it renders

**Comment by jvdp1** [comment]
> An "easy" win I think would be to move the specs under this project. At least to try out and see how it renders

The specs as they are currently are not really useful for the users, at least as an introduction.

See [this comment](https://github.com/fortran-lang/stdlib/discussions/599#discussioncomment-2072353) for some aims.

The curreny [website](https://awvwgk.github.io/stdlib-docs/en/index.html) contains 2 examples ([npy](https://awvwgk.github.io/stdlib-docs/en/how-to/npy.html) and [stats](https://awvwgk.github.io/stdlib-docs/en/how-to/statistics.html)) to illustrate how to use `stdlib`. I think additional examples similar to those should be added.
Specs as they are now can be found in the [Reference page](https://awvwgk.github.io/stdlib-docs/en/spec/index.html) (but I agree that it could be easier to find them).

See also [here](https://github.com/awvwgk/stdlib-docs/issues) from more details on how to build the different sections (and their aims)

**Comment by jalvesz** [comment]
Yes, that's why I mentioned that, when I say "move" the specs I also mean to use the occasion and reshape them under the umbrella of this project. And probably make them.more appealing and easier to use/discover (?)

**Comment by jvdp1** [comment]
Great! I misunderstood your previous comment. Sorry for the confusion.
